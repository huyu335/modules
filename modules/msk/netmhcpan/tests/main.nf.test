nextflow_process {

    name "Test Process NETMHCPAN"
    script "../main.nf"
    process "NETMHCPAN"

    tag "modules"
    tag "modules_nfcore"
    tag "netmhcpan"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("test_neoantigen") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['temp_test_short_maf'], checkIfExists: true),
                    ]

                input[1] = [
                    file(params.test_data_mskcc['neoantigen']['winners_hla_txt'], checkIfExists: true),
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.mutatedSequences).match()
                },
                { assert process.out.xls[0][1] ==~ ".*/test.xls" },
                { assert process.out.netmhcpanoutput[0][1] ==~ ".*/test.netmhcpan.output" }
            )
        }

    }

}
