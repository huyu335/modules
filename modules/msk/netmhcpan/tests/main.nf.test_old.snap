{
    "*_out/*.mutated_sequences.fa": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.mutated_sequences.fa:md5,87c87ae7ccecadfad00e0fc6c4c01498"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-13T17:39:38.583128"
    },
    "test_neoantigen": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.xls:md5,c96616193b944811a46e689480b7f9bb"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.netmhcpan.output:md5,bbc99a8ab3c6b863c2c3f4592b87a9b8"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mutated_sequences.fa:md5,87c87ae7ccecadfad00e0fc6c4c01498"
                    ]
                ],
                "3": [
                    "versions.yml:md5,35ec563839ee27410cf9f8d134c6e8e5"
                ],
                "mutatedSequences": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.mutated_sequences.fa:md5,87c87ae7ccecadfad00e0fc6c4c01498"
                    ]
                ],
                "netmhcpanoutput": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.netmhcpan.output:md5,bbc99a8ab3c6b863c2c3f4592b87a9b8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,35ec563839ee27410cf9f8d134c6e8e5"
                ],
                "xls": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.xls:md5,c96616193b944811a46e689480b7f9bb"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-13T17:19:07.709907"
    }
}